src/main/java/org/prebid/server/bidder/yeahmobi/YeahmobiBidder.java

```diff
@@ -0,0 +1,174 @@
+import org.prebid.server.proto.openrtb.ext.request.yeahmobi.ExtImpYeahmobi;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class YeahmobiBidder implements Bidder<BidRequest> {
+
+    private static final TypeReference<ExtPrebid<?, ExtImpYeahmobi>> YEAHMOBI_EXT_TYPE_REFERENCE =
+            new TypeReference<ExtPrebid<?, ExtImpYeahmobi>>() {
+            };
+
+    private static final String DEFAULT_BID_CURRENCY = "USD";
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+
+    public YeahmobiBidder(String endpointUrl, JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final List<BidderError> errors = new ArrayList<>();
+        final List<HttpRequest<BidRequest>> result = new ArrayList<>();
+
+        for (Imp imp : request.getImp()) {
+            try {
+                final ExtImpYeahmobi extImpYeahmobi = parseImpExt(imp);
+                final Imp processImp = processImp(imp);
+                final String url = endpointUrl + "gw-" + extImpYeahmobi.getZoneId() + "-bid.yeahtargeter.com";
+                result.add(createSingleRequest(processImp, request, url));
+            } catch (PreBidException e) {
+                errors.add(BidderError.badInput(e.getMessage()));
+            }
+        }
+
+        return Result.of(result, errors);
+    }
+
+    private ExtImpYeahmobi parseImpExt(Imp imp) {
+        try {
+            return mapper.mapper().convertValue(imp.getExt(), YEAHMOBI_EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(e.getMessage(), e);
+        }
+    }
+
+    private Imp processImp(Imp imp) {
+
+        final String updateNativeRequest;
+        try {
+            final JsonNode nativeRequest = imp.getXNative() != null
+                    ? mapper.mapper().readValue(imp.getXNative().getRequest(), JsonNode.class)
+                    : null;
+
+            final ObjectNode objectNode = mapper.mapper().createObjectNode().set("native", nativeRequest);
+            updateNativeRequest = nativeRequest != null && nativeRequest.get("native") == null
+                    ? mapper.mapper().writeValueAsString(objectNode)
+                    : null;
+        } catch (JsonProcessingException e) {
+            throw new PreBidException(e.getMessage(), e);
+        }
+
+        return updateNativeRequest != null
+                ? imp.toBuilder().xNative(Native.builder().request(updateNativeRequest).build()).build()
+                : imp;
```

Comment:
They have different logic in GO. 
We need to work with json that in `native.request`.
```
  "native": {
          "request": "{\"native\":{}}\"
}
```
So if there are no imp.native we do not need to do anything. 
If there are no imp.native.request JSON NATIVE we need to add this object. You can check their tests to fix our IT tests.